src/TestFramework/TestFramework/Attributes/TestMethod/TimeoutAttribute.cs

```diff
@@ -19,9 +19,15 @@ public sealed class TimeoutAttribute : Attribute
     /// <param name="timeout">
     /// The timeout in milliseconds.
     /// </param>
-    public TimeoutAttribute(int timeout)
+    /// <param name="cleanupTimeout">
+    /// The cleanup timeout in milliseconds.
+    /// This represents the time allowed for the cleanup (TestCleanup, Dispose) to complete when the test times out.
+    /// The default value is infinite, meaning that all cleanup will be called and left infinite time to process.
+    /// </param>
+    public TimeoutAttribute(int timeout, int cleanupTimeout = int.MaxValue)
```

Comment:
I have made multiple tests and I think it's better to have the default behavior being "call cleanup".

Reply:
agree